kube-runtime/src/watcher.rs

```diff
@@ -805,15 +830,17 @@ pub fn watch_object<K: Resource + Clone + DeserializeOwned + Debug + Send + 'sta
     name: &str,
 ) -> impl Stream<Item = Result<Option<K>>> + Send {
     watcher(api, Config::default().fields(&format!("metadata.name={name}"))).map(|event| match event? {
-        Event::Deleted(_) => Ok(None),
+        Event::Deleted(_) | Event::RestartedStart | Event::RestartedDeleted(_) | Event::RestartedDone => {
```

Comment:
```suggestion
        Event::Deleted(_) | Event::RestartedDeleted(_) => {
```
i think we only want to return a None here if the object explicitly was deleted. a relist here does not indicate this, unless i am missing something.

Reply:
I think you probably need to filter out these events from this function.

Reply:
Hopefully this fixes it: https://github.com/kube-rs/kube/pull/1494/commits/40d71ae50cd40724458d16b4f768a0224c3599c0

Reply:
Fixed in: https://github.com/kube-rs/kube/pull/1494/commits/608ddb2c27d40675dd3533a55c3291c1603f90ce